dev-client/src/hooks/connectivityHooks.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useContext} from 'react';
+
+import {ConnectivityContext} from 'terraso-mobile-client/context/ConnectivityContext';
+
+export const useIsOffline = () => {
+  const context = useContext(ConnectivityContext);
+
+  if (context === null) {
+    console.warn(
+      'We expect to have set up the isOffline state before using this hook',
+    );
+    return null;
```

Comment:
The null check feels a bit clunky. Is there a better way to enforce that we expect the `isOffline` state is created (by the `ConnectivityContextProvider`) before anything else uses the `ConnectivityContext`?

Reply:
Instead of `createContext<ConnectivityContextType | null>(null);` with a nullable context value, you could just set the type to  `SiteUserRole` and have the initial value be some default `ConnectivityContextType` object. (Basically the null-object pattern rather than actual `null`). Not sure what the implications would be of assuming that the context should be online / offline in its default uninitialized state, but presumably it'd be easy to settle on a sensible convention.